src/NexusMods.App.UI/Windows/MainWindowViewModel.cs

```diff
@@ -49,11 +47,12 @@ public class MainWindowViewModel : AViewModel<IMainWindowViewModel>, IMainWindow
             serviceProvider: serviceProvider
         );
 
-        topBarViewModel.AddPanelDropDownViewModel = new AddPanelDropDownViewModel(WorkspaceController);
-        TopBar = topBarViewModel;
+        TopBar = serviceProvider.GetRequiredService<ITopBarViewModel>();
+        TopBar.AddPanelDropDownViewModel = new AddPanelDropDownViewModel(WorkspaceController);
+
+        Spine = serviceProvider.GetRequiredService<ISpineViewModel>();
+        DevelopmentBuildBanner = serviceProvider.GetRequiredService<IDevelopmentBuildBannerViewModel>();
```

Comment:
Why not getting these through constructor injection instead of `GetRequiredService()`?

Reply:
This way they get instantiated at the `GetRequiredService` call instead before the constructor. If they get instantiated before the constructor, then the `MainWindowViewModel` won't be able to register itself as a new window.

Reply:
Ah, I see. This might be worth adding a comment to then.